pkg/provisioner/crdprovisioner.go

```diff
@@ -251,37 +257,64 @@ func (c *CrdProvisioner) CreateVolume(
 		}()
 		return nil, err
 	}
+	azVolumeInstance = obj.(*v1alpha1.AzVolume)
 
 	return azVolumeInstance.Status.Detail.ResponseObject, nil
 }
 
 func (c *CrdProvisioner) DeleteVolume(ctx context.Context, volumeID string, secrets map[string]string) error {
-	azV := c.azDiskClient.DiskV1alpha1().AzVolumes(c.namespace)
-
 	// TODO: Since the CRD provisioner needs to the AzVolume name and not the ARM disk URI, it should really
 	// return the AzVolume name to the caller as the volume ID. To make this work, we would need to implement
 	// snapshot APIs through the CRD provisioner.
 	// Replace them in all instances in this file.
+	lister := c.conditionWatcher.informerFactory.Disk().V1alpha1().AzVolumes().Lister().AzVolumes(c.namespace)
+	azVolumeClient := c.azDiskClient.DiskV1alpha1().AzVolumes(c.namespace)
+
 	volumeName, err := azureutils.GetDiskNameFromAzureManagedDiskURI(volumeID)
 	if err != nil {
-		klog.Errorf("Invalid diskURI (%s) for DeleteVolume operation. Error : (%v)", volumeID, err)
+		klog.Errorf("Error finding volume : %v", err)
 		return nil
 	}
-
 	azVolumeName := strings.ToLower(volumeName)
 
-	azVolume, err := azV.Get(ctx, azVolumeName, metav1.GetOptions{})
+	waiter, err := c.conditionWatcher.waitForObjectCondition(ctx, azvolume, azVolumeName, func(obj interface{}) (bool, error) {
+		// if no object is found, object is deleted
+		if obj == nil {
+			return true, nil
+		}
+		// otherwise, get object from lister
+		obj, err := lister.Get(azVolumeName)
```

Comment:
Perhaps `conditionFunc` could take an additional Boolean or event type parameter to discern a delete rather than performing a cached lookup?